include/parameter.h

```diff
@@ -0,0 +1,307 @@
+                std::string error = get_type_name() + ".set_limits(" + limits->str()
+                    + ") sets limits that are less restrictive than the minimum="
+                    + limits_maximal.str();
+                throw std::runtime_error(error);
+            }
+            _limits_ptr = std::move(limits);
+            _limiter = std::make_unique<Limiter>(*_limits_ptr);
+
+        }
+    }
+    void set_modifiers(const SetC modifiers)
+    {
+        if(modifiers.empty()) _modifiers = nullptr;
+        else {
+            _modifiers = std::make_unique<SetC>();
+            for(auto & modifier : modifiers) _modifiers->insert(modifier);
+        }
+    }
+    void set_transform(const std::shared_ptr<const Transform<T>> transform) {
+        if(transform == nullptr) {
+            _transformer = std::make_unique<Transformer>(transform_none);
+        } else {
+            _transform_ptr = std::move(transform);
+            _transformer = std::make_unique<Transformer>(*_transform_ptr);
+        }
+        _value_transformed = _transformer->transform.forward(_value);
+    }
+
+    void set_value(T value) {
+        _set_value(value);
+        double value_new = this->get_value();
+        _value_transformed = _transformer->transform.forward(value_new);
+        if(_inheritors != nullptr)
+        {
+            for(auto & inheritor : *_inheritors) inheritor->set_value(value_new);
+        }
+    };
+    
+    void set_value_transformed(T value_transformed) {
+        _set_value(_transformer->transform.reverse(value_transformed));
+        _value_transformed = _transformer->transform.forward(this->get_value());
+        if(_inheritors != nullptr)
+        {
+            for(auto & inheritor : *_inheritors) inheritor->set_value_transformed(_value_transformed);
+        }
+    }
+
+    void set_unit(std::shared_ptr<const Unit> unit = nullptr) {
+        _unit_ptr = unit == nullptr ? nullptr : std::move(unit);
+    }
+
+    std::string str() const {
+        return get_type_name() + "(" + std::to_string(_value) + "," + get_limits().str() + ")";
+    }
+
+    static constexpr const Limits<T> limits_maximal = Limits<T>(get_min(), get_max(),
+        type_name<C>(), ".limits_maximal");
+    static constexpr const Transform<T> & transform_none = get_transform_unit<T>();
+
+    Parameter(
+        T value = get_default(),
+        std::shared_ptr<const Limits<T>> limits = nullptr,
+        const std::shared_ptr<const Transform<T>> transform = nullptr,
+        std::shared_ptr<const Unit> unit = nullptr,
+        bool fixed = false,
+        std::string label = "",
+        const SetC inheritors = {},
+        const SetC modifiers = {}
+    ) : ParameterBase<T, Unit>() {
+        set_limits(limits);
+        _value = value;
+        set_transform(transform);
+        // inheritors need to go before value, which is set by set_transform
+        set_inheritors(inheritors);
```

Comment:
```suggestion
        set_inheritors(std::move(inheritors));
```
Or pass by const reference.  More important here than for `shared_ptr`, since the copy penalty for `shared_ptr` is just an extra atomic refcount increment/decrement pair.

Compiler might optimize this double-copy away anyhow, but better to guarantee that since it's easy to do.

Lots of other cases of this you should look for; I certainly haven't flagged them all.